### PR TITLE
gh-115775: Add whatsnew entry about __static_attributes__

### PR DESCRIPTION
**Pull request by iritkatriel**
* Issue: gh-115775

----
📚 Documentation preview 📚: https://cpython-previews--117909.org.readthedocs.build/

### CODE REVIEW

File: Doc/whatsnew/3.13.rst [hunk 1]
```diff
@@ -115,6 +115,10 @@ Improved Error Messages
         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^
     TypeError: split() got an unexpected keyword argument 'max_split'. Did you mean 'maxsplit'?
 
+* Classes have a new ``__static_attributes__``, populated by the compiler,
```
Comment (Eclips4) [inline Doc/whatsnew/3.13.rst]:
```suggestion
* Classes have a new ``__static_attributes__`` attribute, populated by the compiler,
```

**Comment by encukou** [comment]
Thanks for the description!

I'm not sure what the use case is for adding this as public API. But since it is exposed, I guess it should also be added to the [list of special attributes](https://docs.python.org/3/reference/datamodel.html#custom-classes).

(Sorry about the drip-feed of suggestions :( I'm not following the change closely -- I'm just reacting to the effect on projects tested with the latest alpha.)

File: Doc/reference/datamodel.rst [hunk 2]
```diff
@@ -1000,6 +1001,10 @@ Special attributes:
       A tuple containing the :ref:`type parameters <type-params>` of
       a :ref:`generic class <generic-classes>`.
 
+   :attr:`__static_attributes__`
+      A tuple containing names of attributes of this class which are accessed
+      through ``self.X`` from any function in its body.
+
```
Comment (iritkatriel) [inline Doc/reference/datamodel.rst:1007]: The boss said I can ignore this warning: 
image

**Comment by encukou** [comment]
AFAIK, the missing definition should be added like this: https://github.com/iritkatriel/cpython/pull/57

File: Doc/whatsnew/3.13.rst [hunk 3]
```diff
@@ -115,6 +115,11 @@ Improved Error Messages
         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^
     TypeError: split() got an unexpected keyword argument 'max_split'. Did you mean 'maxsplit'?
 
+* Classes have a new ``__static_attributes__`` attribute, populated by the compiler,
```
Comment (JelleZijlstra) [inline Doc/whatsnew/3.13.rst]:
```suggestion
* Classes have a new :attr:`~class.__static_attributes__` attribute, populated by the compiler,
```

File: Doc/library/stdtypes.rst [hunk 4]
```diff
@@ -5542,6 +5542,13 @@ types, where they are relevant.  Some of these are not reported by the
       [<class 'bool'>, <enum 'IntEnum'>, <flag 'IntFlag'>, <class 're._constants._NamedIntConstant'>]
 
 
+.. method:: class.__static_attributes__
```
Comment (JelleZijlstra) [inline Doc/library/stdtypes.rst]:
```suggestion
.. attribute:: class.__static_attributes__
```

**Comment by encukou** [comment]
Thank you!

**Pull request merged**